### DISCUSSION TITLE
Tasmota editor for rules

### DISCUSSION THREAD

**Discussion by dusan06** [arendst/Tasmota#11700] [Q&A] [answered]
Can someone help me! I am trying to use TDM (Tasmota device manager) NODEMCA ESP8266 with 32 pins, when I open the RULES program cracks. Does anyone have an experience or idea.

Tasmota editor for rules, something like TDM (not work with NODEMCU ESP8266)

**Comment by barbudor** [comment, marked as answer]
Hi @dusan06 
I can reproduce the problem.
Please open an Issue on https://github.com/jziolkowski/tdm
But jziolkowski is quite busy currently and may not be able to look at it very quickly
I will try to have a look tonight
In the meantime, you may want to look at [TasmoManager](https://github.com/arendst/Tasmota/discussions/11509) but I'not sure if there is already a rule editor
Alternatively, the console works fine. I often prepare my rules in a text editor then copy to the console after removing the line jumps

**Reply by barbudor** [reply]
@dusan06 Sorry for late feedback, 
It appears that the MQTT message in response to a `Rule` command has changed 10 days ago which breaks TDM parsing.
I will look into this and propose a PR to jziolkowski

**Reply by barbudor** [reply]
@dusan06 I have made a PR for fixing this issue https://github.com/jziolkowski/tdm/pull/138
JZ will merge this morning

**Reply by barbudor** [reply]
Merge done, new exe will be available during the week-end
